### DISCUSSION TITLE
How to read file in Next.js

### DISCUSSION THREAD

**Discussion by lancejpollard** [vercel/next.js#58217] [Help]
### Summary

I have this code in one of my `pages/**/*.tsx`:

```ts
import CWD from '~/tools/cwd'

export async function getServerSideProps({
  query,
  resolvedUrl,
}: GetServerSidePropsContext): Promise<
  GetServerSidePropsResult<FlowPageType>
> {
  let intro = null
  try {
    intro = await serialize(
      fs.readFileSync(
        `${CWD}/contents/language/${query.slug}/Intro.mdx`,
        'utf-8',
      ),
    )
  } catch (e) {}

  // ...
}
```

The `contents` folder is at the top-level.

Then I have that `tools/cwd` in the top-level tools folder:

```ts
import { dirname, resolve } from 'path'
import { fileURLToPath } from 'url'

const __dirname = dirname(fileURLToPath(import.meta.url))

const CWD = resolve(`${__dirname}/..`)

export default CWD
```

That works fine locally when I run `next dev`, but it doesn't work in production on Vercel. What am I missing, what should change?

That is, the `intro` is a valid value locally, but is undefined in production, because apparently the file path is incorrect or something.

**Comment by icyJoseph** [comment]
GSSP, runs in a task, which is not collocated with the source code, so you need to use `process.cwd()` IIRC, to point back to the working directory of the node process.

- https://nextjs.org/docs/pages/api-reference/functions/get-static-props#reading-files-use-processcwd
